### ISSUE TITLE
JS bundle isn't included in Android production build with Gradle 3.4

### ISSUE THREAD

**Issue by thib92** [facebook/react-native#24968]
React Native version: `0.59.8`

## Steps To Reproduce

1. Open Android-Studio and update `com.android.tools.build:gradle` to `3.4.1` in the main Gradle file
2. Sync Gradle
3. Try to build an APK and install it on an Android Simulator (`adb install`)
4. See the app crash
5. Run `adb logcat` to get the logs of the Android Simulator
6. See the exception saying `Unable to load script from assets 'index.android.bundle'`

I encountered the issue and it was fixed by downgrading to `3.3` (see [this reply](https://github.com/facebook/react-native/issues/18357#issuecomment-491782270)).

## Reason of the issue

The explanation in [this comment](https://github.com/facebook/react-native/issues/18357#issuecomment-437186504):

> Apparently, Gradle 3.2.0 changed the location to which it generates asset bundles. RN expects its JS asset bundle to be in a specific path, which is the path used by Gradle 3.1.4. Because G3.2.0 puts them in a different spot, RN cannot find the JS asset bundle, and thus the files with the Javascript to run. Ergo, the crash in release mode.

## Possible fix

I think it would be a good idea to change the default JS bundle path in `react.gradle`.

WDYT? If this is a good fix, I'd be happy to try and implement it.

**Comment by wildseansy** [comment]
My workaround for this was to move the `index.android.bundle` from `app/build/generated` to `app/src/main/assets`, and then run gradle again to bundle this in.

Do this at your own risk, it definitely took some pain to figure this out, and may have unforeseen consequences

I run this to generate a release:
```bash
# Run in android directory:
rm -rf ./app/build/outputs/apk/production
rm -rf ./app/build/generated/assets
./gradlew assembleRelease
cp ./app/build/generated/assets/react/production/release/index.android.bundle ./app/src/main/assets/index.android.bundle
./gradlew assembleRelease
```

**Comment by thib92** [comment]
@wildseansy I’ve seen this workaround as well, but I’m not a big fan as you’re adding a bundle to the src directory, which is in source control, which mean you have to modify your gitignore file etc.

**Comment by wildseansy** [comment]
@thib92 - for sure. I just haven't found a better way yet, and downgrading will come at a cost after having integrated some React Hooks.

**Comment by thib92** [comment]
I use hooks in my app and downgrading to `Gradle 3.1.4` didn't cause any problem 😊 
But anyway I think it would be great if this could be fixed on the RN side!

**Comment by wildseansy** [comment]
Ah, I see - I had assumed downgrading Gradle would mean also bumping RN down, which would cascade to React being bumped down.

**Comment by thib92** [comment]
Fortunately it doesn't look like it 😄

**Comment by guhungry** [comment]
Duplicated with https://github.com/facebook/react-native/issues/24854

**Comment by thib92** [comment]
@guhungry I didn’t see it had been fixed on master! Thanks!

**Comment by guhungry** [comment]
It was fixed in 0.59 so you have to upgrade React Native or search for workaround which i can't remember where i found it.
